content/Properties/Properties.q.md

```diff
@@ -0,0 +1,1077 @@
+
+ - It replaces the string `Canada` of the `address` property with `$pObject`.
+
+:::
+
+
+:::
+
+/// type=REPL
+
+```php
+<?php
+    class Person 
+    {
+        public $name = "Diana";
+        public $address;
+        
+        public function eat()
+        {
+            echo "This is an eat() method.";
+        }
+    }
+    $pObject = new Person();
+    $pObject->address = "Canada";
+    echo $pObject->address;
+?>
+```
+/// type=SS, answer=[2]
+
+On line 5, replace the property definition `public $address;` with `public $address = "California";`. Execute the program. What is its output?
+
+ - It prints `Diana`.
+
+ - It prints `Canada`.
+
+ - It prints `California`.
+
+ - No output is displayed.
+
+ - It prints `This is an eat() method.`.
+
+:::
+
+
+:::
+
+/// type=REPL
+
+```php
+<?php
+    class Person 
+    {
+        public $name = "Diana";
+        public $address = "California";
+        
+        public function eat()
+        {
+            echo "This is an eat() method.";
+        }
+    }
+    $pObject = new Person();
+    $pObject->address = "Canada";
+    echo $pObject->address;
+?>
+```
+/// type=SS, answer=[3]
+
+Remove the statement `$pObject->address = "Canada";` on line 13. Execute the program. What is its output?
+
+ - It prints `Diana`.
+
+ - It prints `Canada`.
+
+ - It prints `California`.
+
+ - No output is displayed.
+
+ - It prints `This is an eat() method.`.
+
+:::
+
+
+:::
+
+/// type=REPL
+
+```php
+<?php
+    class Person 
+    {
+        public $name = "Diana";
+        public $address = "California";
+        
+        public function eat()
+        {
+            echo "This is an eat() method.";
+        }
+    }
+    $pObject = new Person();
+    echo $pObject->address;
+?>
+```
+/// type=SS, answer=[1]
+
+In the statement `echo $pObject->address;` on line 13, replace `address` with `name`. Execute the program. What is its output?
+
+ - It prints `Diana`.
+
+ - It prints `Canada`.
+
+ - It prints `California`.
+
+ - No output is displayed.
+
+ - It prints `This is an eat() method.`.
+
+:::
+
+
+:::
+
+/// type=REPL
+
+```php
+<?php
+    class Person 
+    {
+        public $name = "Diana";
+        public $address = "California";
+        
+        public function eat()
+        {
+            echo "This is an eat() method.";
+        }
+    }
+    $pObject = new Person();
+    echo $pObject->name;
+?>
+```
+/// type=SS, answer=[5]
+
+On line 13, replace the statement `echo $pObject->name;` with `$pObject->eat();`. Execute the program. What is its output?
+
+ - It prints `Diana`.
+
+ - It prints `Canada`.
+
+ - It prints `California`.
+
+ - No output is displayed.
+
+ - It prints `This is an eat() method.`.
+
+:::
+
+
++++
+
+
++++
+
+### Part 2: Knowledge Assessment
+
+/// type=SS, answer=[3]
+
+Which statement best describes a property?
+
+ - It is the value of a class.
+
+ - It is an instance of a class.
+
+ - It is a variable defined inside a class.
```

Comment:
Properties are **named members** of classes, structures, and interfaces. Member variables or methods in a class or structures are called **Fields**. 

Students might be confused if they find another defined variable inside the class functions if we use the term `variable` for the properties.

Reply:
Got your point here... but I think this is already understood by the learners since we have mentioned that it is inside the class and not inside the methods of the class. We also have to make sure that our description here are easy to understand learners without any plagiarism issues. I used these as my reference.
![image](https://user-images.githubusercontent.com/25945790/62585947-f1807b80-b8ed-11e9-92f3-b9a8fbeb73ca.png)
![image](https://user-images.githubusercontent.com/25945790/62585996-27256480-b8ee-11e9-992d-3946370e1273.png)

Reply:
We can use easy to understand description as long as we don't sacrifice the terms used in the development. 
This is ok. You can leave it as is.